core/src/main/java/org/testcontainers/utility/MountableFile.java

```diff
@@ -301,7 +301,14 @@ private void recursiveTar(String entryFilename, String rootPath, String itemPath
             final File sourceRootFile = new File(rootPath).getCanonicalFile();     // e.g. /foo
             final String relativePathToSourceFile = sourceRootFile.toPath().relativize(sourceFile.toPath()).toFile().toString();    // e.g. /bar/baz
 
-            final TarArchiveEntry tarEntry = new TarArchiveEntry(sourceFile, entryFilename + "/" + relativePathToSourceFile); // entry filename e.g. /xyz/bar/baz
+            final String tarEntryFilename;
+            if (relativePathToSourceFile.isEmpty()) {
+                tarEntryFilename = entryFilename; // entry filename e.g. xyz => xyz
```

Comment:
I wonder what will happen if the user will call with it `/some/entry`? 
Maybe we should strip `/` prefix from `tarEntryFilename` before passing to `TarArchiveEntry`'s constructor, for the additional safety

Reply:
Hmm, I think it's allowed, isn't it? There's no source 'working directory' in this context, so `'abc' == '/abc'`..?

I'll give it a try though.